java/cuvs-java/src/main/java/com/nvidia/cuvs/CagraGraph.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.cuvs;
+
+import com.nvidia.cuvs.spi.CuVSProvider;
+
+/**
+ * Represent a CAGRA graph backed by off-heap memory.
+ *
+ * @since 25.08
+ */
+public interface CagraGraph extends AutoCloseable {
```

Comment:
The more I look at this, the more I think "what if we collapse all into `Dataset`"?

What if `Dataset` becomes an interface to construct, carry around and access off-heap matrices (of a set of primitive types we need, e.g. starting with 8 and 32 bit integers and 32 bit floats?)

We can then provide uniform utility functions to e.g. copy them to on-heap arrays, wrap them in DLTensors, etc.

For now just an idea I'm kicking around to see if it makes sense.

Reply:
I've implemented this in the last commit, and I think it's a powerful interface. I'll extract the changes in a separate PR so it can be reviewed separately, but it looks to work very well in this case.

Reply:
I second your idea for a better name. I think we could have a "Matrix" or similar data structure that represents "something that holds 2D data whenever they are". 
Then if we feel we need to make a distinction, we can have lightweight Dataset/Graph classes that use it.